### ISSUE TITLE
[Bug]: Building GraphAr C++ failed on ARM platform

### ISSUE THREAD

**Issue by acezen** [alibaba/GraphAr#157]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

when building GraphAr C++, got error:
```
enumerator value ‘-1’ is outside the range of underlying type ‘char’
```

https://github.com/alibaba/GraphAr/blob/main/cpp/include/gar/writer/arrow_chunk_writer.h#L41-L46

### Expected Behavior

Building without error.

### Minimal Reproducible Example

```bash
cd cpp
cmake ..
make
```

### Environment

- Operating system: ARM
- GraphAr version: v0.5

**Comment by acezen** [comment]
hi, @lixueclaire, could you take a look to the issue, it seems that some arm architectures would take `char` as `unsigned`
https://developer.arm.com/documentation/den0013/d/Porting/Miscellaneous-C-porting-issues/unsigned-char-and-signed-char

**Comment by lixueclaire** [comment]
fixed by #158
